### DISCUSSION TITLE
Unable to choose device for IP address

### DISCUSSION THREAD

**Discussion by vrastiak** [netbox-community/netbox#14814] [Q&A] [answered]
Hi,
I am on 3.7.0.. I am not sure what went wrong when but now I cannot choose device for IP address in order to assign it to the interface. I cannot choose it for existing IP nor newly added IP address. I can see only interface selection as if there was some device already preselected. Has anyone had similar issue?

![obrazek](https://github.com/netbox-community/netbox/assets/156896953/1ef5df31-0264-4662-9569-25c1be547690)

**Comment by markkuleinio** [comment, marked as answer]
There is a button in the right of the Interface list.

**Comment by vrastiak** [comment]
Right, thank you, I feel stupid now :)

**Reply by markkuleinio** [reply]
No probs, I don't know why there even is the interface dropdown list shown at all, it's very rare to have an interface name that is NOT also found in some other device, so why not just show the button. Maybe that's a limitation of Django, I don't know.

Ah, obviously it's there to show the interface name AFTER it has been already selected.

**Reply by candlerb** [reply]
This is a case where the old UI (select device first, then select interface) was more logical and far easier to use. It would be fine to have a search button next to "Device" so you can filter it to devices in a particular site, rack, tenant etc. But every interface *must* be attached to a Device, and interface names are typically only unique within a device.
